### PR TITLE
Update release notes and add deprecation notices for SelectMenu and Dropdown

### PR DESCRIPTION
**Pull request by rezrah**
Follow up to https://github.com/primer/react/pull/1881

- adds deprecation annotations to missing components
  
  Screenshot 2022-02-24 at 13 10 22

- updates release notes

### CODE REVIEW

File: .changeset/smooth-cameras-prove.md [hunk 1]
```diff
@@ -1,5 +1,84 @@
...
+    </ActionList>
+  </ActionMenu.Overlay>
+</ActionMenu>
+```
+
+</td>
+</tr>
+</table>
+
+See [https://primer.style/react/ActionMenu](https://primer.style/react/ActionMenu) for more migration examples.
```
Comment (rezrah) [inline .changeset/smooth-cameras-prove.md:60]: Pointing at the old ActionMenu for now, but once this is merged, this should be the correct one 🤞

File: docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ export default function resolveScope(metastring) {
     ...doctocatComponents,
     ...primerComponents,
     ...(metastring.includes('drafts') ? drafts : {}),
-    ...deprecated,
+    ...(metastring.includes('deprecated') ? deprecated : {}),
```
Comment (siddharthkp) [inline docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js:55]: We will also need to add `deprecated` to the examples for deprecated components

```diff
+ ```jsx live deprecated
<BorderBox>This is a BorderBox</BorderBox>
```

File: docs/content/deprecated/Dropdown.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ status: Deprecated
 
 ## Deprecation
 
-Use [DropdownMenu](/DropdownMenu) instead.
+Use [ActionMenu](/ActionMenu) instead.
```
Comment (siddharthkp) [inline docs/content/deprecated/Dropdown.md:8]: 👍

File: docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js (hunk 2 above)
Comment (siddharthkp) [inline docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js:55]: This is good! 

We will also need to add `deprecated` to the examples for deprecated components

```diff
+ ```jsx live deprecated
<BorderBox>This is a BorderBox</BorderBox>
```

**Review by siddharthkp — CHANGES_REQUESTED** [review]
We will also need to add deprecated to the examples for deprecated components

```diff
+ ```jsx live deprecated
<BorderBox>This is a BorderBox</BorderBox>
```

File: docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js (hunk 2 above)
Comment (rezrah) [reply docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js:55]: done @siddharthkp

File: .changeset/smooth-cameras-prove.md [hunk 4]
```diff
@@ -1,5 +1,269 @@
 ---
-"@primer/react": major
+'@primer/react': major
 ---
 
-Move deprecated components to deprecated folder
+### SelectMenu
+
+> ⚠️ `SelectMenu` has been deprecated. Please use `ActionMenu` instead.
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:7]:
```suggestion
⚠️ `SelectMenu` has been deprecated. Please use `ActionMenu` instead.
```

File: .changeset/smooth-cameras-prove.md [hunk 5]
```diff
@@ -1,5 +1,269 @@
...
+
+<table>
+<tr>
+<th> Before </th> <th> After </th>
+</tr>
+<tr>
+<td valign="top">
+
+```jsx
+// import {SelectMenu} from '@primer/react';
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:17]: Are imports intentionally commented out?

File: .changeset/smooth-cameras-prove.md [hunk 6]
```diff
@@ -1,5 +1,269 @@
...
+
+</td>
+</tr>
+</table>
+
+See [https://primer.style/react/ActionMenu](https://primer.style/react/ActionMenu) for more migration examples.
+
+### Dropdown
+
+> ⚠️ `Dropdown` has been deprecated. Please use `ActionMenu` instead.
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:62]:
```suggestion
⚠️ `Dropdown` has been deprecated. Please use `ActionMenu` instead.
```

File: .changeset/smooth-cameras-prove.md [hunk 7]
```diff
@@ -1,5 +1,269 @@
...
+
+<table>
+<tr>
+<th> Before </th> <th> After </th>
+</tr>
+<tr>
+<td valign="top">
+
+```jsx
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:71]: These examples don't have imports. I recommend we have imports in all of them or none of them.

File: .changeset/smooth-cameras-prove.md [hunk 8]
```diff
@@ -1,5 +1,269 @@
...
+
+</td>
+</tr>
+</table>
+
+See [https://primer.style/react/ActionMenu](https://primer.style/react/ActionMenu) for more migration examples.
+
+### Flex
+
+> ⚠️ `Flex` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:125]:
```suggestion
⚠️ `Flex` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```

File: .changeset/smooth-cameras-prove.md [hunk 9]
```diff
@@ -1,5 +1,269 @@
...
+</Box>
+```
+
+</td>
+</tr>
+</table>
+
+### BorderBox
+
+> ⚠️ `BorderBox` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:203]:
```suggestion
⚠️ `BorderBox` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```

File: .changeset/smooth-cameras-prove.md [hunk 10]
```diff
@@ -1,5 +1,269 @@
...
+</Box>
+```
+
+</td>
+</tr>
+</table>
+
+### Position
+
+> ⚠️ `Position` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```
Comment (mperrotti) [inline .changeset/smooth-cameras-prove.md:235]:
```suggestion
⚠️ `Position` has been deprecated. Please use [`Box`](https://primer.style/react/Box) instead.
```

File: docs/content/deprecated/BorderBox.md [hunk 11]
```diff
@@ -12,21 +12,21 @@ Use [Box](/Box) instead.
 
 **Before**
 
-```jsx
+```jsx deprecated
```
Comment (mperrotti) [inline docs/content/deprecated/BorderBox.md:15]: Love this addition to doctocat <3

**Review by mperrotti — APPROVED** [review]
Approved assuming @siddharthkp's feedback gets addressed. Just some non-blocking nitpicks and commit suggestions.

**Comment by mperrotti** [comment]
Are we intentionally keeping the components in `src` instead of `src/deprecated/`?

File: .changeset/smooth-cameras-prove.md (hunk 7 above)
Comment (rezrah) [reply .changeset/smooth-cameras-prove.md:67]: removed them, thanks for reminder.

**Review by siddharthkp — APPROVED** [review]
ship it :)

**Pull request merged**